### DISCUSSION TITLE
Binding Grid.RowDefinitions / Grid.ColumnDefinitions

### DISCUSSION THREAD

**Discussion by DM-3** [AvaloniaUI/Avalonia#17820] [Q&A] [answered]
Trying to bind the RowDefinitions and ColumnDefinitions properties of a grid to ViewModel class gives error AVLN3000.
Binding other properties of the same object works as intended.
Is it intentional that these properties cannot be bound?

axaml snippet in question:
```XAML
<Grid
    RowDefinitions="{Binding GridRows}"
    ColumnDefinitions="{Binding GridColumns}"
/>
```

cs snippet in question:
```C#
public RowDefinitions GridRows { get; } = new RowDefinitions();
public ColumnDefinitions GridColumns { get; } = new ColumnDefinitions();
```

**Comment by stevemonaco** [comment, marked as answer]
The binding system requires you to bind target properties that are `StyledProperty` or `DirectProperty`. `Grid.RowDefinitions` is a normal CLR property. Similar to WPF, it's not designed to be binded to and mutating can be difficult. I tend to use `Grid` as a fixed table layout, although not fixed cell size. However, that's not strictly the case if you're mutating in code-behind.
